spec.md

```diff
@@ -2644,7 +2801,10 @@ implies `hash(x) == hash(y)`.
 `int(x[, base])` interprets its argument as an integer.
```

Comment:
> a simple polynomial accumulator over the UTF-16 transcoding of the string

Ug. What will this do to compatibility with implementations not using UTF-16?

Reply:
It does this:
https://github.com/google/starlark-go/blob/master/starlark/library.go#L408

Reply:
Hm. Seems like the Java interpreter is really throwing its weight around. Compare with what we did for strings, where neither implementation is favored over the other.

Why is it necessary that the hash of strings be fully specified across implementations, when their length and the bitsize of their elements is not?

Reply:
In the course of building Herb, I encountered .bzl files whose effects were exquisitely dependent on the numbers returned by explicit `hash(str)` calls. Using an equally valid but different hash function changed execution so much that it exposed latent bugs in those files, or created significantly different build targets. This was my reluctant solution to the problem.

Reply:
That doesn't explain why hashes need to be specified across implementations of the interpreter. We don't need Bazel .bzl files to be processed by the Go implementation of Starlark. And even if we do, we can leave it out of the spec so that we're a little freer to change it later (at risk of practical incompatibilities from inertia).

Reply:
On the contrary, it's a compelling reason why to specify hashes: unless the hash functions match exactly, Herb cannot simulate Blaze, and there is no workaround.

Reply:
It can't do it anyway if the string implementation is different, can it?